src/vl53l0x_driver.cpp

```diff
@@ -1,185 +1,141 @@
+                              1);
+  VL53L0X_SetLimitCheckEnable(pDevice,
+                              VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE, 1);
+  VL53L0X_SetLimitCheckValue(pDevice,
+                             VL53L0X_CHECKENABLE_SIGNAL_RATE_FINAL_RANGE,
+                             (FixPoint1616_t)(0.1 * 65536));
+  VL53L0X_SetLimitCheckValue(pDevice, VL53L0X_CHECKENABLE_SIGMA_FINAL_RANGE,
+                             (FixPoint1616_t)(60 * 65536));
+  VL53L0X_SetMeasurementTimingBudgetMicroSeconds(pDevice, 33000);
+  VL53L0X_SetVcselPulsePeriod(pDevice, VL53L0X_VCSEL_PERIOD_PRE_RANGE, 18);
+  VL53L0X_SetVcselPulsePeriod(pDevice, VL53L0X_VCSEL_PERIOD_FINAL_RANGE, 14);
 }
 
-void Start_Ranging() {
-    VL53L0X_PerformSingleRangingMeasurement(pSensorOne,	&SensorOneRangingMeasurementData);
-    VL53L0X_PerformSingleRangingMeasurement(pSensorTwo,	&SensorTwoRangingMeasurementData);
-    VL53L0X_PerformSingleRangingMeasurement(pSensorThree, &SensorThreeRangingMeasurementData);
-    VL53L0X_PerformSingleRangingMeasurement(pSensorFour, &SensorFourRangingMeasurementData);
+void Start_Ranging(int i) {
 
-    sensorData.One = SensorOneRangingMeasurementData.RangeMilliMeter;
-    sensorData.Two = SensorTwoRangingMeasurementData.RangeMilliMeter;
-    sensorData.Three = SensorThreeRangingMeasurementData.RangeMilliMeter;
-    sensorData.Four = SensorFourRangingMeasurementData.RangeMilliMeter;
+  VL53L0X_PerformSingleRangingMeasurement(pSensors[i],
+                                          &SensorsRangingMeasurementData[i]);
+  sensor_msg_array[i].data = SensorsRangingMeasurementData[i].RangeMilliMeter;
 }
 
 int main(int argc, char **argv) {
-    ros::init(argc, argv, "vl53l0x_driver");
-    ros::NodeHandle nh;
-    ros::Publisher pub = nh.advertise<vl53l0x_driver::vl53l0x>("/vl53l0x_sensors_data", 1);
-
-    GPIO_Setup();
-    Sensor_Setup();
-
-    //Step 4 :: Calibration Sensor and Get Sensor Value 
-    Sensor_Calibration(pSensorOne);
-    Sensor_Calibration(pSensorTwo);
-    Sensor_Calibration(pSensorThree);
-    Sensor_Calibration(pSensorFour);
-
-    while (ros::ok()) {
-        Start_Ranging();
-        pub.publish(sensorData);
-        ros::spinOnce();
+  initialize();
+
+  ros::init(argc, argv, "vl53l0x_driver");
+  ros::NodeHandle nh;
+  nh.getParam("/measure_proximity_node/i2c_bus_instance", i2c_bus_instance);
+  ROS_INFO("i2c_bus_instance: %d", i2c_bus_instance);
+  i2c_bus_path = "/dev/i2c-" + std::to_string(i2c_bus_instance);
+
+  ros::Publisher sensor_pub_array[NUM_SENSORS];
+  std::string name = "sensor_data_";
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    std::string result = name + std::to_string(i + 1);
+    sensor_pub_array[i] = nh.advertise<std_msgs::Int16>(result, 1000);
+  }
+
+  GPIO_Setup();
+  Sensor_Setup();
+
+  // Step 4 :: Calibration Sensor and Get Sensor Value
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    Sensor_Calibration(pSensors[i]);
+  }
+  while (ros::ok()) {

```

Comment:
Also check if i2c device file is available before publishing:
e.g.:
```suggestion
check_device_connection(int _file_descriptor) {
    struct stat device_status;
    fstat(_file_descriptor, &device_status);
    return device_status.st_nlink;
}
  while (ros::ok() && check_device_connection(pSensors[0]->fd)) {
```

Reply:
ok